src/patches/vault-historic-data-refetch.ts

```diff
@@ -0,0 +1,104 @@
+import { ChartTimeFrame } from '@badger-dao/sdk';
+
+import { getDataMapper, getVaultEntityId } from '../aws/dynamodb.utils';
+import { ChartDataBlob } from '../aws/models/chart-data-blob.model';
+import { HistoricVaultSnapshotModel } from '../aws/models/historic-vault-snapshot.model';
+import { VaultDefinitionModel } from '../aws/models/vault-definition.model';
+import { getSupportedChains } from '../chains/chains.utils';
+import { toChartDataKey } from '../charts/charts.utils';
+import { getEnvVar } from '../config/config.utils';
+import { VaultStrategy } from '../vaults/interfaces/vault-strategy.interface';
+import { getStrategyInfo } from '../vaults/vaults.utils';
+
+/**
+ * Run Patch
+ * npx ts-node -r dotenv/config src/patches/vault-historic-data-refetch.ts
+ *
+ * Target Patch
+ * STAGE=staging | prod
+ * VAULT="0x..."
+ *
+ * Requires DDB write access + MFA authentication to patch.
+ *
+ * Due IO errors in indexer, default vaules could be saved in DDB and corrapt data in charts
+ * This patch will update the charts with the correct data.
+ *
+ * Data to be updated:
+ * - `strategy` field
+ *
+ * https://github.com/Badger-Finance/badger-api/issues/1539
+ */
+
+async function vaultHistoricDataRefetch() {
+  console.log('Patch:VaultSnapshotRefetch - Started');
+
+  getEnvVar('STAGE');
+
+  const vaultAddress = getEnvVar('VAULT');
+
+  const mapper = getDataMapper();
+
+  for (const chain of getSupportedChains()) {
+    let vault: VaultDefinitionModel;
+
+    try {
+      vault = await chain.vaults.getVault(vaultAddress);
+    } catch (err) {
+      console.log(`Patch:VaultSnapshotRefetch - 
+      Vault ${vaultAddress} not found for chain ${chain.network}. Next tick...`);
+      continue;
+    }
+
+    let cachedChart: ChartDataBlob<HistoricVaultSnapshotModel> | undefined;
+    const vaultEntityId = getVaultEntityId(chain, vault);
+
+    let strategyInfo: VaultStrategy | undefined;
+
+    try {
+      strategyInfo = await getStrategyInfo(chain, vault);
```

Comment:
hmm, should we utilize the `blockTag` on entries? maybe it is ok not to as it does not change very often for this purpose (none changed during the issue period)

Reply:
ohh shoot, good point ty